staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -40,6 +50,40 @@ func handleError(w http.ResponseWriter, r *http.Request, err error) {
 	glog.Errorf(err.Error())
 }
 
+// requestWatermark is used to trak maximal usage of inflight requests.
+type requestWatermark struct {
+	lock                                 sync.Mutex
+	readOnlyWatermark, mutatingWatermark int
+}
+
+func (w *requestWatermark) record(readOnlyVal, mutatingVal int) {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+
+	if w.readOnlyWatermark < readOnlyVal {
+		w.readOnlyWatermark = readOnlyVal
+	}
+	if w.mutatingWatermark < mutatingVal {
+		w.mutatingWatermark = mutatingVal
+	}
+}
+
+var watermark requestWatermark
+
+func startRecordingUsage() {
+	go func() {
+		wait.Forever(func() {
+			watermark.lock.Lock()
+			defer watermark.lock.Unlock()
+			metrics.UpdateInflightRequestMetrics(watermark.readOnlyWatermark, watermark.mutatingWatermark)
```

Comment:
we're holding a lock hanging all requests at this point... either read into local vars or use atomic.ReadInt64/atomic.StoreInt64 to avoid locking here

Reply:
I'd need to both atomically read and write those values. I'm not sure it's worth it.

Reply:
Just read them locally to keep the critical section small before calling Prometheus 

lock
read
reset
unlock
pass to Prometheus

Reply:
Yup, I was going to do this yesterday, but life attacked me before I was able to.